### PR TITLE
atari code

### PR DESCRIPTION
**Pull request by maxtaylordavies**
This PR implements most of the code required for training on atari games using [google's offline RL datasets](https://github.com/google-research/batch_rl). There's also some stuff in here that isn't atari-specific, namely code for running automated sample efficiency evaluation during training (plots of return and episode length against number of samples seen by the model). 

Been having a bit of a nightmare on account of various factors... 
- took me a while to figure out how to load dopamine datasets into our own `CustomDataset` structures. 
- then i was initially doing all evaluation through `gymnasium` envs, same way as we do for minigrid/minari and it wouldn't work - eventually realised that the original DT authors have a custom eval setup using the `atari-py` library, and `atari-py` envs are somehow meaningfully different from the gym/gymnasium envs for atari games??? 🤦‍♂️ 
- then i was still getting shit returns and couldn't figure out why for ages, until i realised that the google DQN data **is not sampled from an expert policy after training, it's sampled from the model throughout training (at every epoch)** - since each epoch of 50 has 1000000 samples, by loading 100k samples i was only getting data from the first 0.2% of training, hence it was crap. now sampling backwards from the final epoch instead of forwards from the first. clearer documentation on this point would be great but probably also my fault for being dumb 

ANYWAY now the cluster seems very congested so i'm having a hard time getting any training jobs to run, but creating this PR now anyway because i don't want to get too far adrift of master.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,7 @@ src/temp.py
 src/old_*
 scripts/baseline_ids.json
 *.json
+.DS_Store
 scripts/test.py
 build/
 test.py
```
Comment (sabraaap) [inline .gitignore:20]: This is missing the sandpit.py from my commit (now in master branch)

File: src/dataset/custom_dataset.py [hunk 2]
```diff
@@ -56,6 +57,30 @@ def save(self):
...
+
+    @property
+    def level_name(self):
+        return self._level_name
+
+    @property
+    def missions(self):
+        return self._missions
+
+    @property
```
[the remark this answers is not in the document]
Comment (maxtaylordavies) [reply src/dataset/custom_dataset.py]: oops thanks

**Review by sabraaap — CHANGES_REQUESTED** [review]
Wow that looks like a lot of work, great effort! Could you please update the .gitignore file and the custom_dataset file according to my comments? Maybe also generally double check  custom_dataset against the master branch, just because some changes from my PR (now in master) were not on your branch yet :-)

**Comment by maxtaylordavies** [comment]
> Wow that looks like a lot of work, great effort! Could you please update the .gitignore file and the custom_dataset file according to my comments? Maybe also generally double check custom_dataset against the master branch, just because some changes from my PR (now in master) were not on your branch yet :-)

cheers, will do!

**Review by sabraaap — APPROVED** [review]
Looking good! 💃

**Pull request merged**